sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/CatalystQlSuite.scala

```diff
@@ -170,4 +171,10 @@ class CatalystQlSuite extends PlanTest {
     parser.parsePlan("select sum(product + 1) over (partition by (product + (1)) order by 2) " +
       "from windowData")
   }
+
+  test("string charset literal") {
+    compareExpressions(
+      parser.parseExpression("'ISO-8859-1' 0x4341464562616265"),
```

Comment:
MySQL apparently supports this: https://docs.oracle.com/cd/E17952_01/refman-5.5-en/charset-literal.html